conf/machine/imx8mnul-ddr3l-evk.conf

```diff
@@ -4,7 +4,11 @@
 #@DESCRIPTION: Machine configuration for NXP i.MX 8M Nano UltraLite Evaluation Kit with DDR3L
 #@MAINTAINER: Jun Zhu <junzhu@nxp.com>
 
-MACHINEOVERRIDES =. "imx-boot-container:mx8mnul:"
+MACHINEOVERRIDES =. "mx8mnul:"
+
+# FIXME: u-boot-imx should be converted to `binman` and then we can
+# avoid this specific overrides and handle it in a generic way.
+MACHINEOVERRIDES =. "${@bb.utils.contains('IMX_DEFAULT_BOOTLOADER', 'u-boot-imx', '', 'imx-boot-container:', d)}"
```

Comment:
I prefer to handle this generically inside the class itself. 

We have a variable called `UBOOT_PROVIDES_BOOT_CONTAINER` to control this. Setting this inside U-Boot i.MX fork should solve this globally.

Could you give it a try?

Reply:
I don't understand. The override `imx-boot-container` is used to set `UBOOT_PROVIDES_BOOT_CONTAINER`, so this PR simply aligns the new machine config with the other machine configs:

```
conf/machine/imx8mq-evk.conf:MACHINEOVERRIDES =. "${@bb.utils.contains('IMX_DEFAULT_BOOTLOADER', 'u-boot-imx', '', 'imx-boot-container:', d)}"
conf/machine/include/imx8mn-evk.inc:MACHINEOVERRIDES =. "${@bb.utils.contains('IMX_DEFAULT_BOOTLOADER', 'u-boot-imx', '', 'imx-boot-container:', d)}"
conf/machine/include/imx8mm-evk.inc:MACHINEOVERRIDES =. "${@bb.utils.contains('IMX_DEFAULT_BOOTLOADER', 'u-boot-imx', '', 'imx-boot-container:', d)}"
conf/machine/include/imx8mp-evk.inc:MACHINEOVERRIDES =. "${@bb.utils.contains('IMX_DEFAULT_BOOTLOADER', 'u-boot-imx', '', 'imx-boot-container:', d)}"
```

Reply:
Do you mean you want to rework that so we no longer rely on the override `imx-boot-container` for setting `UBOOT_PROVIDES_BOOT_CONTAINER`?

Reply:
Yes. Take a look at #1465. I made the PR cleaning it.